src/app/api/tasks/tasks.service.ts

```diff
@@ -84,22 +96,46 @@ export class TasksService extends BaseService {
     // while clients can only view the tasks assigned to them or their company
     const filters = this.buildReadFilters(id)
 
-    return await this.db.task.findFirst({
+    const task = await this.db.task.findFirst({
       ...filters,
       include: {
         workflowState: true,
       },
     })
+    if (!task) throw new APIError(httpStatus.NOT_FOUND, 'The requested task was not found')
+
+    return task
   }
 
   async updateOneTask(id: string, data: UpdateTaskRequest) {
     const policyGate = new PoliciesService(this.user)
     policyGate.authorize(UserAction.Update, Resource.Tasks)
 
-    return await this.db.task.update({
+    // Query previous task
+    const filters = this.buildReadFilters(id)
+    const prevTask = await this.db.task.findFirst({
+      ...filters,
+      include: { workflowState: true },
+    })
+    if (!prevTask) throw new APIError(httpStatus.NOT_FOUND, 'The requested task was not found')
+
+    // Get the updated task
+    const updatedTask = await this.db.task.update({
```

Comment:
as part of the this update, I think we should track `completedAt` date property to know when this task was completed. And similarly for assignedAt. These important state transitions will be helpful in the future.

Reply:
Fixed in a separate PR: https://github.com/copilot-platforms/tasks-app/pull/24

Reply:
We will be merging this branch for now and implementing these changes in new PRs because they're quite heavy